transaction/_transaction.py

```diff
@@ -760,3 +753,13 @@ def __init__(self, datamanager):
 
     def rollback(self):
         raise TypeError("Savepoints unsupported", self.datamanager)
+
+def text_or_warn(s):
+    if isinstance(s, text_type):
+        return s
+
+    warnings.warn("Expected text", DeprecationWarning, stacklevel=3)
+    if isinstance(s, bytes):
+        return s.decode('utf-8', 'replace')
+    else:
+        return text_type(s)
```

Comment:
RelStorage didn't check for None before it ran `str`, so if `user` and `description` were ever set to that (which seems likely, especially for `user`), there would be `'None'` in the database.

My point is, it probably doesn't matter that much 😄